spec/models/time_period_spec.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: time_periods
+#
+#  id         :bigint           not null, primary key
+#  end_date   :date
+#  start_date :date
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+require 'rails_helper'
+
+RSpec.describe TimePeriod, type: :model do
+  let!(:time_period3) { FactoryBot.create(:time_period, start_date: Date.current - 8.weeks, end_date: Date.current - 7.weeks) }
+  let!(:time_period2) { FactoryBot.create(:time_period, start_date: Date.current - 6.weeks, end_date: Date.current - 5.weeks) }
+  let!(:time_period1) { FactoryBot.create(:time_period, start_date: Date.current - 4.weeks, end_date: Date.current - 3.weeks) }
+  let!(:time_period) { create :time_period }
+
+  context 'Validations' do
+    subject { FactoryBot.build(:time_period) }
+    it { is_expected.to validate_presence_of(:start_date) }
+    it { is_expected.to validate_presence_of(:end_date) }
+
+    it 'fails when end_date earlier then start_date' do
+      time_period.end_date = time_period.start_date - 1.day
+      expect(time_period).to_not be_valid
+    end
+
+    it 'fails when start_date earlier then the latest end_date' do
+      time_period.start_date = Date.current - 4.weeks
+      expect(time_period).to_not be_valid
+    end
+  end
+
+  context '.create_time_period' do
+    it 'should create new time period record' do
+      current_day = Date.current.beginning_of_week(:sunday)
+      expect{TimePeriod.create_time_period}.to change { TimePeriod.all.count }.by(1)
+      expect(TimePeriod.last.start_date).to eq current_day
+      expect(TimePeriod.last.end_date).to eq current_day + 6.days
```

Comment:
Please have the `TimePeriod.create_time_period` return the newly created object. After that you can test it like so:
```ruby
new_time_period = TimePeriod.create_time_period
expect(new_time_period.start_date).to eq(last_sunday)
```